docs/content/concepts/sui-move-concepts/conventions.mdx

```diff
@@ -0,0 +1,331 @@
+
+  // ❌ Wrong
+  let name2 = profile.profile_age();
+}
+```
+## Empty Function
+
+#### Functions that create data structures must be called empty.
+
+```move
+module conventions::data_structure {
+  struct DataStructure has copy, drop, store {
+    bits: vector<u8>
+  }
+
+  public fun empty(): DataStructure {}
+}
+```
+## New Function
+
+#### Functions that create objects must be called new.
+
+```move
+module conventions::object {
+  struct Object has key, store {
+    id: UID
+  }
+
+  public fun new(ctx:&mut TxContext): Object {}
+}
+```
+
+## Shared Objects
+
+#### Shared objects must be created via a new function and be shared in a separate function. The share function must be named share.
+
+```move
+module conventions::profile {
+  struct Profile has key {
+    id: UID
+  }
+
+  public fun new(ctx:&mut TxContext): Profile {}
+
+  public fun share(profile: Profile) {}
+}
+```
+
+## Reference Functions
+
+#### Functions that return a reference must be named property_name_mut or property_name_immut. The borrow is implied by the mut and immut.
+
+```move
+module conventions::profile {
+  struct Profile has key {
+    id: UID,
+    name: String,
+    age: u8
+  }
+
+  // profile.name_immut()
+  public fun name_immut(self: &Profile): &String {}
+
+  // profile.age_mut()
+  public fun age_mut(self: &mut Profile): &mut u8 {}
+}
+```
+
+## Separation of Concerns
+
+#### Modules must be designed around one Object or Data Structure. A variant structure should have its own module to avoid complexity and bugs.
+
+```move
+module conventions::wallet {
+  struct Wallet has key, store {
+    id: UID,
+    amount: u64
+  }
+}
+
+module conventions::claw_back_wallet {
+  struct Wallet has key {
+    id: UID,
+    amount: u64
+  }
+}
+```
+
+## Errors
+
+#### Errors must be CamelCase, start with an E and be descriptive.
+
+```move
+module conventions::errors {
+  // ✅ Right
+  const ENameHasMaxLengthOf64Chars: u64 = 0;
+
+  // ❌ Wrong
+  const INVALID_NAME: u64 = 0;
+}
+```
+
+## Struct Property Comments
+
+#### Describe the properties of your structs.
+
+```move
+module conventions::profile {
+  struct Profile has key, store {
+    id: UID,
+    // The age of the user
+    age: u8,
+    // The first name of the user
+    name: String
+  }
+}
+```
+
+## Destroy Functions
+
+#### Provide functions to delete objects and structs. Empty objects must be destroyed with the function destroy_empty. The function drop must be used for objects that have types that can be dropped.
+
+```move
+module conventions::wallet {
+  struct Wallet<Value> {
+    id: UID,
+    value: Value
+  }
+
+  // Value has drop
+  public fun drop<Value: drop>(wallet: &mut Wallet<Value>) {}
+
+  // Value doesn't have drop
+  // Does NOT throw if the `wallet` is not empty.
+  public fun destroy<Value>(wallet: &mut Wallet<Value>) {}
+
+  // Value doesn't have drop
+  // Throws if the `wallet` is not empty.
+  public fun destroy_empty<Value>(wallet: &mut Wallet<Value>) {}
```

Comment:
I think I see the point you're making here, but the example you gave may not be the best one. Destroy empty is a necessity for droppable collection types if the value doesn't have drop. Here Wallet is not droppable and not a collection.